Bug: pre-commit hooks failing during local execution

**Describe the bug**
The local `pre-commit` workflow fails on Linux during the `cppcheck` and `unit-test` stages. While `clang-format` passes, the subsequent steps prevent successful commits due to pathing issues and C++ standard mismatches.

### Error Logs

#### 1. cppcheck failure
The hook fails because it cannot locate `.c` files in the zoslib test directory:
```text
clang++: error: no such file or directory: '/home/yinjie/SEAR/tests/zoslib/*.c'
```

#### 2. unit-test failure
The compiler fails to recognize `std::string_view`, which is required by conversion.hpp:
```test
/home/yinjie/SEAR/sear/conversion.hpp:8:55: error: no type named 'string_view' in namespace 'std'
8 | std::string toUTF8(const std::string& input, std::string_view codepage = "IBM-1047");
```

**Root Cause Analysis**
- Pathing issue `cppcheck` : 
In the `Makefile`, `SRCZOSLIB` is defined as `$(ZOSLIB)/*.c` for linux and mac, but the directory contains no `.c` files (only `.cpp` or is empty), causing the compiler to error out when the wildcard fails to expand.

- Version Mismatch `unit-test`:
The Makefile explicitly sets `CXXSTANDARD = c++14`. However, `conversion.hpp` utilizes `std::string_view`, a feature introduced in C++17. This prevents compilation on environments that strictly enforce the specified standard.


**Environment Information:**

- OS : Linux
- Compiler: Clang++
- C++ Standard targeted: C++17

**Additional context**
I'm happy to submit a PR to update the `Makefile` and refine the hooks to ensure they are robust across different local environments. This will help maintain code quality and catch these errors before they reach the CI.

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -8,7 +8,11 @@ repos:
     hooks:
       - id: cppcheck
         name: cppcheck
-        entry: make check
+        entry: | 
+          bash -c '
+            cmake --preset linux-test
+            cmake --build --preset linux-test --target check
+          '
         language: system
         pass_filenames: false
         always_run: true
@@ -17,7 +21,13 @@ repos:
     hooks:
       - id: unit-test
         name: unit-test
-        entry: make test
+        entry: | 
+          bash -c '
+            cmake --preset linux-test -DBUILD_SHARED_LIBS=on
+            cmake --build --preset linux-test --target sear
+            cmake --build --preset linux-test --target test_runner
+            ./build/linux/test_runner
+          '
         language: system
         pass_filenames: false
         always_run: true

--- a/Makefile
+++ b/Makefile
@@ -18,7 +18,7 @@ TESTS			= ${PWD}/tests
 ZOSLIB			= $(TESTS)/zoslib
 
 CSTANDARD		= c99
-CXXSTANDARD		= c++14
+CXXSTANDARD		= c++17
 
 COMMON_INC		= \
 				-I $(SRC) \
@@ -28,8 +28,8 @@ COMMON_INC		= \
 				-I $(KEY_MAP) \
 				-I $(VALIDATION) \
 				-I $(JSON) \
-				-I $(JSON_SCHEMA)
-				-I $(ICONV) \
+				-I $(JSON_SCHEMA) \
+				-I $(ICONV)
 
 # JSON Schemas
 SEAR_SCHEMA	= $(shell cat ${PWD}/schema.json | jq -c)
@@ -68,7 +68,7 @@ else ifeq ($(UNAME), Darwin)
 	CC			= clang
 	CXX			= clang++
 
-	SRCZOSLIB	= $(ZOSLIB)/*.c
+	SRCZOSLIB	= $(ZOSLIB)/*.cpp
 
 	CFLAGS		= \
 				-std=$(CXXSTANDARD) -D__ptr32= \
@@ -88,7 +88,7 @@ else
 	CC			= clang
 	CXX			= clang++
 
-	SRCZOSLIB	= $(ZOSLIB)/*.c
+	SRCZOSLIB	= $(ZOSLIB)/*.cpp
 
 	CFLAGS		= \
 				-std=$(CXXSTANDARD) -D__ptr32= \

